pkg/shortnames/shortnames.go

```diff
@@ -328,21 +328,26 @@ func Resolve(ctx *types.SystemContext, name string) (*Resolved, error) {
 	// If we're running in disabled, return the candidates without
 	// prompting (and without recording).
 	if mode == types.ShortNameModeDisabled {
+		if len(unqualifiedSearchRegistries) == 0 {
+			return nil, errors.Errorf("short-name %q did not resolve to an alias and no unqualified-search registries are defined in %q", name, usrConfig)
+		}
 		return resolved, nil
 	}
 
 	// If we have only one candidate, there's no ambiguity. In case of an
 	// empty candidate slices, callers can implement custom logic or raise
 	// an error.
-	if len(resolved.PullCandidates) <= 1 {
+	if len(resolved.PullCandidates) == 1 {
 		return resolved, nil
 	}
 
 	// If we don't have a TTY, act according to the mode.
 	if !terminal.IsTerminal(int(os.Stdout.Fd())) || !terminal.IsTerminal(int(os.Stdin.Fd())) {
 		switch mode {
 		case types.ShortNameModePermissive:
-			// Permissive falls back to using all candidates.
+			if len(unqualifiedSearchRegistries) == 0 {
+				return nil, errors.Errorf("short-name %q did not resolve to an alias and no unqualified-search registries are defined in %q", name, usrConfig)
```

Comment:
> ```go
> // PullCandidates in the returned value may
> // be empty if there is no matching alias and no unqualified-search registries
> // are configured.
> ```

This is working as documented. If that doesn’t make sense, should we change the API now while it’s early enough that we can pretend it’s not a break?

Reply:
Good catch.  Yes, I think we should change the doc comments. IMHO it's a bug fix.

Reply:
I’d like to go back to the earlier discussions, this was definitely _intentional_ but I can’t remember whether there was a specific rationale.

Implementation-wise, please make this a single check immediately after getting `unqualifiedSearchRegistries`.

Reply:
As far as I can trace this back, https://github.com/containers/image/pull/1061#discussion_r517645860 , which is no longer applicable now that `ResolveLocally` exists.

So, ACK to this change.

Reply:
Done :heavy_check_mark: